main/src/ca/uwaterloo/flix/runtime/shell/Shell.scala

```diff
@@ -172,17 +180,57 @@ class Shell(sourceProvider: SourceProvider, options: Options) {
   /**
     * Reloads every source path.
     */
-  private def execReload()(implicit terminal: Terminal): Validation[CompilationResult, CompilationMessage] = {
+  private def execReload()(implicit terminal: Terminal): Unit = {
 
     // Scan the disk to find changes, and add source to the flix object
     sourceFiles.addSourcesAndPackages(flix)
     
     // Remove any previous definitions, as they may no longer be valid against the new source
     clearFragments()
 
-    val result = compile(progress = isFirstCompile)
+    compile(progress = isFirstCompile)
     isFirstCompile = false
-    result
+  }
+
+  /**
+    * Displays documentation for the given identifier
+    */
+  private def execDoc(s: String)(implicit terminal: Terminal): Unit = {
+    val w = terminal.writer()
+    val classSym = Symbol.mkClassSym(s)
+    val defnSym = Symbol.mkDefnSym(s)
+    val enumSym = Symbol.mkEnumSym(s)
+    val aliasSym = Symbol.mkTypeAliasSym(s)
+
+    for(r <- root) {
```

Comment:
I would prefer if this was an explicit match on whether root is None or Some :) (for clarity)